### PR TITLE
Barrel-o-fixes

### PR DESCRIPTION
**Pull request by cevich**

### CODE REVIEW

File: files/slave.dockerfile [hunk 1]
```diff
@@ -2,21 +2,20 @@ FROM docker.io/stackbrew/centos:latest
 MAINTAINER Chris Evich <cevich@redhat.com>
 
 RUN yum update -y && \
-    yum install -y findutils epel-release https://rdoproject.org/repos/rdo-release.rpm && \
+    yum install -y epel-release https://rdoproject.org/repos/rdo-release.rpm && \
```
Comment (edsantiago) [inline files/slave.dockerfile]: Why are you removing findutils? I wonder if this is what's making `run_exec/exec_pid_count` fail on fedora?

Comment (cevich) [inline files/slave.dockerfile]: Nope, I just moved it into ``files/slave.rpms``.  This won't affect the test VMs, remember slave == kommandir-container (and needs to be renamed).  i.e. when everything is up and going, the topology looks like this:

```
executing node (e.g. Jenkins) -> kommandir vm -> kommandir container -> peons (test VMs)
```

**Comment by cevich** [comment]
@edsantiago here...maybe I better just annotate the diffs for ya...

File: files/slave.dockerfile (hunk 1 above)
Comment (cevich) [inline files/slave.dockerfile]: Moved ``findutils`` into ``slave.rpms`` because that's where it should be.

File: files/slave.dockerfile [hunk 2]
```diff
@@ -2,21 +2,20 @@ FROM docker.io/stackbrew/centos:latest
...
     # Keep every layer as clean (i.e. small) as possible
     yum clean all
 
-# N/B: Use Ansible 2.1.0 until 2.1.2 is available b/c of:
-#      https://github.com/ansible/ansible/issues/15915
-
```
Comment (cevich) [inline files/slave.dockerfile]: Reverted, we're going the other way now - rpm version of 2.2.1 to source version of 2.2.1 (which is newer)

File: files/slave.dockerfile [hunk 3]
```diff
@@ -2,21 +2,20 @@ FROM docker.io/stackbrew/centos:latest
...
 
 RUN xargs -a /root/slave.rpms yum install -y && \
+    # Keep every layer as clean (i.e. small) as possible
     rm -f /root/slave.rpms && \
     rm -rf /usr/src /usr/share/doc && \
     yum clean all
 
-RUN yum install -y python-pip python-devel && \
+# Development jibblets not needed at runtime, allow later removal
+RUN yum install -y python-pip python-devel openssl openssl-devel libffi libffi-devel; \
```
Comment (cevich) [inline files/slave.dockerfile]: openssl and libffi are needed to build/install ansible from source

File: files/slave.dockerfile [hunk 4]
```diff
@@ -25,13 +24,24 @@ RUN yum install -y python-pip python-devel && \
     yum groupinstall -y 'Development Tools' && \
     # Very specific version is required for older Openstack Env.
     pip install --no-cache-dir --disable-pip-version-check shade==1.12.1 && \
+
+    # Use Ansible from source until rpm available w/ fix for:
+    # https://github.com/ansible/ansible/issues/15915
+    pip install --upgrade setuptools && \
```
Comment (cevich) [inline files/slave.dockerfile]: The ``python-setuptools`` installed from  ``slave.rpms`` is too old, temporarily bump it up using pip (eek!). Effectively, the install in ``slave.rpms`` is just to make sure as many dependencies come in rpm-format as possible (so pip ignores them).

File: files/slave.dockerfile [hunk 5]
```diff
@@ -25,13 +24,24 @@ RUN yum install -y python-pip python-devel && \
...
+    git clone https://github.com/ansible/ansible.git \
+        --branch=stable-2.2 --recursive --depth=0 --single-branch \
+        /usr/src/ansible && \
+    cd /usr/src/ansible && \
+    make install && \
```
Comment (cevich) [inline files/slave.dockerfile]: This actually ends up calling pip to do more dependency fullfilment, but hopefully we satisfied most of them via rpms (above).

File: files/slave.dockerfile [hunk 6]
```diff
@@ -25,13 +24,24 @@ RUN yum install -y python-pip python-devel && \
...
+    
     # Keep every layer as clean (i.e. small) as possible
-    yum history -y undo $(cat /root/trans_num) && \
+    yum history -y undo $(cat /root/trans_num) ; \
```
Comment (cevich) [inline files/slave.dockerfile]: This keeps the kommandir-container (aka slave) image small.  There's no need to hang onto all the development bits needed to build ``andible`` and ``shade``.  Containers based on this image are spun up and removed frequently, so keeping the image small improves performance substantially.

File: files/slave.dockerfile [hunk 7]
```diff
@@ -25,13 +24,24 @@ RUN yum install -y python-pip python-devel && \
...
     rm -f /root/trans_num && \
     yum clean all && \
     rm -rf /usr/src /usr/share/doc
 
-RUN echo "" > /etc/ansible/hosts && \
+RUN mkdir -p /etc/ansible && \
+    echo "" > /etc/ansible/hosts && \
```
Comment (cevich) [inline files/slave.dockerfile]: The previous rpm-based install of ansible too care of making this directory, now we gotta do it here instead.

File: group_vars/fedoratomic/autotested.yml [hunk 8]
```diff
@@ -16,8 +16,7 @@ docker_autotest:
           dest: "{{ autotest_docker_path }}/config_custom/subtests.ini"
         - src: "{{ custom_control_ini_j2 | default('control.ini.j2') }}"
           dest: "{{ autotest_docker_path }}/config_custom/control.ini"
-    preserve_fqins:
-        - "docer.io/fedora:latest"
+    preserve_fqins: []
```
Comment (cevich) [inline group_vars/fedoratomic/autotested.yml]: woopsie

File: roles/yumrepos/meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,4 @@
+---
+
+dependencies:
+  - role: common
```
Comment (cevich) [inline roles/yumrepos/meta/main.yml]: explicit dependencies are always better than implicit 😀

File: vars/peon_defaults.yml [hunk 10]
```diff
@@ -54,23 +54,6 @@ peon_defaults:
...
-    fedoratomic:
-      version: "24"
-      groups:
-        - fedoratomic
-        - atomic_upgrade
-        - autotested
-        - peons
-
-    fedora:
-      volume: "20"
```
Comment (cevich) [inline vars/peon_defaults.yml]: Technically we could do both 24 and 25 - it just requires changing the names (``fedoratomic -> fedoratomic24 and fedoratomic25``).  But why bother, easier all around to only test one Fedora.

**Comment by cevich** [comment]
Yay!  EEEET VERKS! EEEEER VERKS!    (no more ``to_yaml`` or unknown var: ``adept_path``))

**Comment by cevich** [comment]
Ha!  Only one subtest failure to boot!  Okay, merging...

**Pull request merged**